### ISSUE TITLE
Parquet-based tabular datasets not batched by size in ParallelRunStep

### ISSUE THREAD

**Issue by vla6** [Azure/MachineLearningNotebooks#1334]
It appears that if you create a TabularDataset from a parquet file, and then try to use it in the ParallelRunStep, the data is not batched regardless of the value in mini_batch_size.   This contrasts with CSV, where the data is split up into more than 1 batch for parallelization.

I have an example notebook here: [https://github.com/vla6/Azure_notes/blob/main/parallelrunstep_parquet_batch_issue_di.ipynb]

It seems that maybe it would batch multi-file Parquet data but if the Parquet data is in just 1 file, you always get 1 mini batch no matter the size of the file.  I am using pandas to read and write Parquet and not specifying multiple files.  With Pandas it seems like it partitions by column anyway and the parallel run step generally expects row partitions?  

The difference in parquet vs CSV behavior isn't documented anywhere, and some information I've found seems to strongly imply that the source data in the TabularDataset isn't important. For example, in this documentation: [https://docs.microsoft.com/en-us/azure/machine-learning/how-to-debug-parallel-run-step], it says (under mini_batch_size): "For TabularDataset, it's the size of data...."  and does not mention parquet at all. Also in this press release: [https://techcommunity.microsoft.com/t5/azure-ai/batch-inference-in-azure-machine-learning/ba-p/1417010], it implies Parquet data can be batched.

**Comment by anliakho2** [comment]
@vla6 You are right, this should be documented to avoid surprises like that.
For parquet files batch size would be aligned to a row group size of the files. The actual size of row groups and how many rows groups there are in a single parquet file is actually controlled by the producer of these files. So if you have parquet file with many row groups or many parquet files you should see many mini batches produced, but for the case of a single file and single row group you would get behavior you have observed.

**Comment by vla6** [comment]
Thanks for this reply!  When I found this error I was working with Pandas and just wrote out a dataframe with to_parquet().  I imagine this would be a common process for prepping data prior to this kind of activity.  There might be a way to control the row group size with Pandas but it isn't documented in the standard documentation: 
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_parquet.html .  I've found links that say it might be possible with certain "engines" 

In addition this isn't very convenient because in order to try different batch sizes you'd need to write out many files.  Writing to CSV is probably easier but then you lose certain metadata which would be nice to keep sometimes.
